opcua/common/subscription.py

```diff
@@ -208,39 +208,58 @@ def _get_node(self, nodeid):
             node = Node(self.server, ua.NodeId(nodeid))
         return node
 
-    def _get_filter_from_event_type(self, eventtype):
-        eventtype = self._get_node(eventtype)
+    def _get_filter_from_event_type(self, eventtypes):
+        # FIXME must be a nicer way to make a list out of eventtypes when it's passed
+        # FIXME an integer (such as ua.ObjectIds.BaseEventType)
+        evtypes = []
+        if type(eventtypes) is not list:
+            evtypes.append(eventtypes)
+        else:
+            evtypes = eventtypes
+
+        evtype_node_list = []
+        for evtype in evtypes:
+            evtype_node_list.append(self._get_node(evtype))  # make sure we have a list of Node objects
+
         evfilter = ua.EventFilter()
-        evfilter.SelectClauses = self._select_clauses_from_evtype(eventtype)
-        evfilter.WhereClause = self._where_clause_from_evtype(eventtype)
+        evfilter.SelectClauses = self._select_clauses_from_evtype(evtype_node_list)
+        evfilter.WhereClause = self._where_clause_from_evtype(evtype_node_list)
         return evfilter
 
-    def _select_clauses_from_evtype(self, evtype):
+    def _select_clauses_from_evtype(self, evtypes):
         clauses = []
-        for prop in get_event_properties_from_type_node(evtype):
-            op = ua.SimpleAttributeOperand()
-            op.TypeDefinitionId = evtype.nodeid
-            op.AttributeId = ua.AttributeIds.Value
-            op.BrowsePath = [prop.get_browse_name()]
-            clauses.append(op)
+
+        selected_paths = []
+        for evtype in evtypes:
+            for prop in get_event_properties_from_type_node(evtype):
+                if prop.get_browse_name() not in selected_paths:
+                    op = ua.SimpleAttributeOperand()
+                    op.TypeDefinitionId = evtype.nodeid
+                    op.AttributeId = ua.AttributeIds.Value
+                    op.BrowsePath = [prop.get_browse_name()]
+                    clauses.append(op)
+                    selected_paths.append(prop.get_browse_name())
```

Comment:
it might be that you need to remove the line with op.TypeDefinitionId = evtype.nodeid. It is not necessary and we might get issue for different events